src/components/swap/SwapBox.vue

```diff
@@ -130,7 +135,25 @@ export default defineComponent({
     },
     async swapFrom(model) {
       if (model) {
-        // TODO: Call the method that gets the gas fee and calculate swapTo.value
+        this.hasAllowance = false;
+        let balance;
+        if (model.type === "NATIVE") {
+          if (!model.decimals) {
+            console.error("decimals not defined for model ", model.token);
+          }
+          balance = await this.web3.eth.getBalance(this.account);
+
+          this.swapFrom.balance = new BigNumber(balance).shiftedBy(
+            -model.decimals
+          );
+        } else {
+          const tokenInst = new this.web3.eth.Contract(
+            ERC20_ABI,
+            model.address
+          );
+          balance = await tokenInst.methods.balanceOf(this.account).call();
+          this.swapFrom.balance = this.web3.utils.fromWei(balance);
```

Comment:
Just a note here: We are assuming that the ERC20 has 18 decimals here, implicitly (using fromWei). This will work because the WBTC token in RSK will have 18 decimals, but ideally we shouldn't do this. We should ask the contract for the decimals (check the code of the Anyswap ERC20). Don't change it but keep it in mind.